src/bootstrap/native.rs

```diff
@@ -193,7 +193,7 @@ pub(crate) fn is_ci_llvm_available(config: &crate::config::Config, asserts: bool
         return false;
     }
 
-    if crate::util::CiEnv::is_ci() {
+    if crate::util::CiEnv::is_rust_lang_ci() {
```

Comment:
I don't think this is correct for forks. Say that `llvm_sha == head_sha`. That means that LLVM was modified in this commit (even if it's being built out-of-tree). We don't have enough info here to know whether this is a fork of rust-lang/rust, or whether someone's just downloaded the git history in addition to building from source. To be cautious I think we should avoid trying to use LLVM and force a build from source; people can still unset `GITHUB_ACTIONS` if this behavior is wrong for them.

Reply:
I had indeed missed the check in src/bootstrap/config.rs which enforces stage 2 builds in CI environments, it was the most problematic one for me. As is_ci_llvm_available(), I don't mind leaving it untouched, but rather than undefine GITHUB_ACTIONS, I would prefer adding an environment variable that could force-disable rust-lang CI checks. If you do a search for "is_ci()" you'll find a bunch of other helper functions that check for a CI environment for things that are benign in nature, like colorized output, etc.